### ISSUE TITLE
Relative line numbers should change while folding. 

### ISSUE THREAD

**Issue by domi-eck** [microsoft/vscode#169749]
# Vim like navigation while folding with relative line numbers

This request is related to [#159236](https://github.com/microsoft/vscode/pull/159236)

### In NVim/Vim folded sections are simply treated as lines.
For example one section can be deleted with "dd", if the section is folded.
But more importantley, one can just move over folded sections by simply typing the amount of visible line numbers.
The relative line numbers on the side bar do show the visible line numbers and not the acutal ones,
like you can see in the example.

![grafik](https://user-images.githubusercontent.com/75951708/208931593-cb22f6a5-2e9b-4bbf-b867-4900b8e0a935.png)

### To get Vim like behavior, two things are needed. 
First, folds should not open, if you navigate over it. 
Second, the visible line numbers should be visible, like it was also discussed in  [#159236](https://github.com/microsoft/vscode/pull/159236). 

I think, this would help a lot of Vim users moving from from it to VSCode. 
This is the biggest issue for me, after i switched. 

Thanks and keep up the good work!
Cheers

**Comment by ForgottenUmbrella** [comment]
Code folds won't open when navigating over them if you enable the `vim.foldfix` setting.

Ideally, operations (both destructive and navigational) on folds should act on the entire folded region like in vim, instead of only the first line. That is, folded regions should be treated in every respect as if they were a single line. Then a separate line number option wouldn't be necessary, and neither would the foldfix be. (But my understanding is that the foldfix is a hack due to fundamental differences in vscode vs vim).

**Comment by grezp** [comment]
Is there an update on this feature? It appears that it was enabled on [#157282](https://github.com/microsoft/vscode/issues/157282), but then reverted back. Can we just add a new variable/option to enable vim like relative line numbers? This way it's not a breaking behavior.

**Comment by YouRik** [comment]
I agree that this behaviour is the desired one and should be configurable.
It has been implemented before and reverted. This behaviour should simply be behind a configuration.
